### ISSUE TITLE
regression with the new opaque type generics

### ISSUE THREAD

**Issue by aliemjay** [rust-lang/rust#122307]
The following should compile.
```rust
#![feature(type_alias_impl_trait)]
#![allow(unconditional_recursion)]

type Opaque<'a> = impl Sized + 'a;

fn test<'a>() -> Opaque<'a> {
    let _: () = test::<'a>();
    //~^ ERROR expected generic lifetime parameter, found `'_`
}
```

Regressed in #122103. With the new generics of `Opaque`, the defining use at the `let` statement is `Opaque<'_, 'a> := ()`. Cc @compiler-errors

**Comment by compiler-errors** [comment]
You minimized this from the same issue we have with RPITs, right? Because this code ICEs:

```rust
#![allow(unconditional_recursion)]

fn test<'a>() -> impl Sized + 'a {
    let _: () = test::<'a>();
}
```

**Comment by aliemjay** [comment]
No, I discovered it accidentally by rebasing #116891.

**Comment by compiler-errors** [comment]
Wait, @aliemjay isn't this issue also fixed by #116891? It should be filtering out the uncaptured lifetimes in that PR.
